modular_citadel/code/modules/reagents/reagent container/hypospraymkii.dm

```diff
@@ -2,105 +2,105 @@
+	var/start_vial = /obj/item/reagent_containers/glass/bottle/vial/small
 	var/spawnwithvial = TRUE
-	var/start_vial = null
+	var/inject_wait = 50
+	var/spray_wait = 50
+	var/quickload = FALSE
 
-/obj/item/reagent_containers/hypospray/mkii/brute
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/bicaridine
+/obj/item/hypospray/mkii/brute
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/bicaridine
 
-/obj/item/reagent_containers/hypospray/mkii/toxin
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/antitoxin
+/obj/item/hypospray/mkii/toxin
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/antitoxin
 
-/obj/item/reagent_containers/hypospray/mkii/oxygen
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/dexalin
+/obj/item/hypospray/mkii/oxygen
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/dexalin
 
-/obj/item/reagent_containers/hypospray/mkii/burn
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/kelotane
+/obj/item/hypospray/mkii/burn
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/kelotane
 
-/obj/item/reagent_containers/hypospray/mkii/tricord
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/tricord
+/obj/item/hypospray/mkii/tricord
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/small/preloaded/tricord
 
-/obj/item/reagent_containers/hypospray/mkii/CMO
+/obj/item/hypospray/mkii/CMO
 	name = "hypospray mk.II deluxe"
-	allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/small, /obj/item/reagent_containers/glass/bottle/vial/large)
+	allowed_containers = list(/obj/item/reagent_containers/glass/bottle/vial/tiny, /obj/item/reagent_containers/glass/bottle/vial/small, /obj/item/reagent_containers/glass/bottle/vial/large)
 	icon_state = "cmo2"
-	ignore_flags = 1
-	desc = "The Chief Medical Officer's hypospray is identically functional to the base model, excepting that it can take larger vials in addition to regular sized. It is also able to penetrate harder materials and deliver more reagents per spray."
+	desc = "The Deluxe Hypospray can take larger-size vials. It also acts faster and delivers more reagents per spray."
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/CMO
-	possible_transfer_amounts = list(5,10,15,30,60) //cmo hypo should be able to dump lots into it
-
-/obj/item/reagent_containers/hypospray/mkii/CMO/combat
-	name = "Combat Hypospray mk. II"
-	desc = "A hypospray designed for in-combat situations."
-	loaded_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/combat
-
-/obj/item/reagent_containers/hypospray/mkii/Initialize()
+	start_vial = /obj/item/reagent_containers/glass/bottle/vial/large/preloaded/CMO
+	inject_wait = 20
```

Comment:
should put these into defines instead of magic numbers.
#define HYPO_WAIT_CMO = 20

this is a 2 second delay